CimSweep/ArtifactRetrieval/NetworkProfiles.ps1

```diff
@@ -0,0 +1,134 @@
+                
+            $Parameters = @{
+                Hive = 'HKLM'
+                SubKey = 'SOFTWARE\Microsoft\Windows NT\CurrentVersion\NetworkList\Profiles'
+            }
+    
+            Get-CSRegistryKey @Parameters @CommonArgs | ForEach-Object { 
+                
+                $ObjectProperties = @{}
+                $ObjectProperties['PSComputerName'] = $_.PSComputerName
+
+                Get-CSRegistryValue -Hive $_.Hive -SubKey $_.SubKey @CommonArgs | ForEach-Object { 
+                    
+                    $ValueName = $_.ValueName
+                    $ValueContent = $_.ValueContent
+
+                    switch ($ValueName) {
+                        
+                        { $_ -like "Date*" } {
+                            $BinaryReader = New-Object IO.BinaryReader (New-Object IO.MemoryStream (,$ValueContent))
+                        
+                            $Year = $BinaryReader.ReadInt16()
+                            $Month = $BinaryReader.ReadInt16()
+                            $null = $BinaryReader.ReadInt16() # skip week day
+                            $Day = $BinaryReader.ReadInt16()
+                            $Hour = $BinaryReader.ReadInt16()
+                            $Minute = $BinaryReader.ReadInt16()
+                            $Second = $BinaryReader.ReadInt16()
+                            $Millisecond = $BinaryReader.ReadInt16()
+                        
+                            $BinaryReader.BaseStream.Dispose()
+                            $BinaryReader.Dispose()
+
+                            $Date = New-Object datetime -ArgumentList @($Year, $Month, $Day, $Hour, $Minute, $Second, $Millisecond, 'Utc')
+                            
+                            $ObjectProperties[$ValueName] =  $Date.ToString('o')
+                        }
+                        
+                        'NameType' { 
+                            $ObjectProperties['Type'] = switch ($ValueContent) {
+                                      6 { 'Wired' }
+                                     23 { 'VPN' }
+                                     71 { 'Wireless' }
+                                default { $ValueContent }
+                            }
+                        }
+                        
+                        'Category' { 
+                            $ObjectProperties['Category'] = switch ($ValueContent) {
+                                0 { 'Public' }
+                                1 { 'Private' }
+                                2 { 'Domain' }
+                            }
+                        }
+                        
+                        'Managed' { $ObjectProperties['Managed'] = [bool]$ValueContent }
+                          
+                          default { $ObjectProperties[$ValueName] = $ValueContent }
+                    }
+                }
+                [PSCustomObject]$ObjectProperties
```

Comment:
Ah yeah I see that @EricZimmerman's [AppCompatCacheParser](https://github.com/EricZimmerman/AppCompatCacheParser) doesn't handle Store apps either. Without digging in, I recognize these fields:

* 8664 == X64 architecture
* Microsoft.MSPaint == AppX name
* 8wekyb3d8bbwe == PublisherId
* 000106a66d710000 == Version - 0x0001 (1), 0x06a6 (1702), 0x6d71 (28017), 0x0 (0) == 1.1702.28017.0
* 000a000039d00000 == Version (not sure what this corresponds to) - 10.0.14800.0

The first field might refer to a length. For example, the field will be 0xB when there's a ResourceId included (e.g. neutral).

So you could translate much of this into the PackageFullName: `Microsoft.MSPaint_2.1704.26037.0_x64__8wekyb3d8bbwe`

There are also some that include a ResourceId (e.g. neutral). Therefore, the format would look like: `<Name>_<Version>_<Architecture>_<ResourceId>_<PublisherId>`

You can check your work with:
```powershell
Get-AppxPackage -Name 'AppName'
```